packages/canvas/canvas-renderer/src/CanvasRenderer.ts

```diff
@@ -182,289 +246,113 @@ export class CanvasRenderer extends AbstractRenderer
      */
-    public destroy(removeView?: boolean): void
-    {
-        // call the base destroy
-        super.destroy(removeView);
-
-        this.context = null;
+     public destroy(removeView?: boolean): void
+     {
+         this.runners.destroy.items.reverse();
 
-        this.refresh = true;
+         this.emitWithCustomOptions(this.runners.destroy, {
+             _view: removeView,
+         });
 
-        this.maskManager.destroy();
-        this.maskManager = null;
+         super.destroy();
+     }
 
-        this.smoothProperty = null;
-    }
+     get plugins(): IRendererPlugins
+     {
+         return this._plugin.plugins;
+     }
 
-    /**
+     /**
      * Resizes the canvas view to the specified width and height.
      *
-     * @extends PIXI.AbstractRenderer#resize
+     * @implements PIXI.IRenderer#resize
      *
      * @param desiredScreenWidth - the desired width of the screen
      * @param desiredScreenHeight - the desired height of the screen
      */
-    public resize(desiredScreenWidth: number, desiredScreenHeight: number): void
-    {
-        super.resize(desiredScreenWidth, desiredScreenHeight);
-
-        // reset the scale mode.. oddly this seems to be reset when the canvas is resized.
-        // surely a browser bug?? Let PixiJS fix that for you..
-        if (this.smoothProperty)
-        {
-            this.rootContext[this.smoothProperty] = (settings.SCALE_MODE === SCALE_MODES.LINEAR);
-        }
-    }
+     public resize(desiredScreenWidth: number, desiredScreenHeight: number): void
+     {
+         this._view.resizeView(desiredScreenWidth, desiredScreenHeight);
+     }
 
-    /** Checks if blend mode has changed. */
-    invalidateBlendMode(): void
-    {
-        this._activeBlendMode = this.blendModes.indexOf(this.context.globalCompositeOperation);
-    }
+     /**
+     * Same as view.width, actual number of pixels in the canvas by horizontal.
+     *
+     * @member {number}
+     * @readonly
+     * @default 800
+     */
+     get width(): number
+     {
+         return this._view.view.width;
+     }
+
+     /**
+     * Same as view.height, actual number of pixels in the canvas by vertical.
+     *
+     * @member {number}
+     * @readonly
+     * @default 600
+     */
+     get height(): number
+     {
+         return this._view.view.height;
+     }
+
+     get resolution(): number
+     {
+         return this._view.resolution;
+     }
+
+     get autoDensity(): boolean
+     {
+         return this._view.autoDensity;
+     }
+
+     get view(): HTMLCanvasElement
+     {
+         return this._view.view;
+     }
+
+     get screen(): Rectangle
+     {
+         return this._view.screen;
+     }
+
+     get lastObjectRendered(): IRenderableObject
+     {
+         return this._renderer.lastObjectRendered;
+     }
+
+     get renderingToScreen(): boolean
+     {
+         return this._renderer.renderingToScreen;
+     }
 
```

Comment:
`CanvasRenderer` is missing the getter for `blendModes`, `maskManager`, `refresh`, `rootContext`, `smoothProperty`, `setBlendMode`, `invalidateBlendMode`, `setContextTrasform`

I'm guessing `maskMaanger` should be a getter for `CanvasMaskSystem`?

Reply:
getters added..

Reply:
`blendModes`, `maskManager`, `refresh` are missing documentation from `CanvasRenderer`

`preserveDrawingBuffer` and `useContextAlpha` are missing